app/actions/websocket/users.ts

```diff
@@ -120,3 +123,129 @@ export async function handleStatusChangedEvent(serverUrl: string, msg: WebSocket
     const newStatus = msg.data.status;
     setCurrentUserStatus(serverUrl, newStatus);
 }
+
+export async function handleCustomProfileAttributesValuesUpdatedEvent(serverUrl: string, msg: WebSocketMessage) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        logError('No operator found');
+        return;
+    }
+
+    const {user_id, values} = msg.data;
+    const attributesForDatabase = Object.entries(values).map(([fieldId, value]) => ({
+        id: `${fieldId}-${user_id}`,
+        field_id: fieldId,
+        user_id,
+        value: value as string,
+    }));
+
+    try {
+        await operator.handleCustomProfileAttributes({
+            attributes: attributesForDatabase,
+            prepareRecordsOnly: false,
+        });
+    } catch (error) {
+        logError('Error handling custom profile attributes values updated event', error);
+    }
+}
+
+export async function handleCustomProfileAttributesFieldUpdatedEvent(serverUrl: string, msg: WebSocketMessage) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        logError('No operator found');
+        return;
+    }
+
+    const {field} = msg.data;
+    const fieldForDatabase = {
+        id: field.id,
+        group_id: field.group_id,
+        name: field.name,
+        type: field.type,
+        target_id: field.target_id,
+        target_type: field.target_type,
+        create_at: field.create_at,
+        update_at: field.update_at,
+        delete_at: field.delete_at,
+        attrs: field.attrs,
+    };
```

Comment:
again, said it before but will say it again, this transformation should happen in the operator

Reply:
actually there is no transformatino here anymore, removing this part